vendors/nxp/boards/lpc54018iotmodule/ports/secure_sockets/iot_secure_sockets.c

```diff
@@ -474,6 +474,204 @@ int32_t SOCKETS_Send( Socket_t xSocket,
+                xTempAddress.sin_family = pxAddress->ucSocketDomain;
+                xTempAddress.sin_port = SOCKETS_ntohs( pxAddress->usPort );
+
+                sendBuf = custom_alloc( xDataLength );
+
+                if( sendBuf != NULL )
+                {
+                    memcpy( sendBuf, pvBuffer, xDataLength );
+                    lStatus = qcom_sendto( ( int ) pxContext->xSocket,
+                                           sendBuf,
+                                           xDataLength,
+                                           0,
+                                           ( struct sockaddr * ) &xTempAddress,
+                                           ( int ) xAddressLength );
+                    custom_free( sendBuf );
+
+                    if( lStatus < 0 )
+                    {
+                        /* Set the error code to be returned */
+                        lStatus = SOCKETS_SOCKET_ERROR;
+                    }
+                }
+                else
+                {
+                    /* Error: TX buffer allocation failure */
+                    lStatus = SOCKETS_ENOMEM;
+                }
+            }
+            else
+            {
+                /* Error: Socket closed for write */
+                lStatus = SOCKETS_ECLOSED;
+            }
+        }
+        else
+        {
+            /* Error: Invalid parameter */
+            lStatus = SOCKETS_EINVAL;
+        }
+
+        return lStatus;
+    }
+/*-----------------------------------------------------------*/
+
+    int32_t SOCKETS_RecvFrom( Socket_t xSocket,
+                              void * pvBuffer,
+                              size_t xDataLength,
+                              uint32_t ulFlags,
+                              SocketsSockaddr_t * pxAddress,
+                              Socklen_t * xAddressLength )
+    {
+        TickType_t xTimeOnEntering = xTaskGetTickCount();
+        int32_t lStatus = SOCKETS_ERROR_NONE;
+        SSOCKETContextPtr_t pxContext = ( SSOCKETContextPtr_t ) xSocket;
+        SOCKADDR_T xTempAddress = { 0 };
+        char * buffLoc = NULL;
+        QCA_CONTEXT_STRUCT * enetCtx = wlan_get_context();
+        A_STATUS xSelectStatus;
+
+        if( ( SOCKETS_INVALID_SOCKET != xSocket ) &&
+            ( NULL != pvBuffer ) &&
+            ( NULL != pxAddress ) )
+        {
+            if( !( nxpsecuresocketsSOCKET_READ_CLOSED_FLAG & pxContext->xShutdownFlags ) )
+            {
+                pxContext->xRecvFlags = ( BaseType_t ) ulFlags;
+
+                for( ; ; )
+                {
+                    /* Check if there is anything to be received on this socket. */
+                    xSelectStatus = ( A_STATUS ) t_select( enetCtx,
+                                                           ( uint32_t ) pxContext->xSocket,
+                                                           nxpsecuresocketsONE_MILLISECOND );
+
```

Comment:
Do we have to use select here ? Does NXP support non blocking sockets ? Why can't  we use qcom_recvfrom() directly ?

Reply:
Referring to the doc, 3.2.2 UDP receive "recvfrom call is blocking, so select can be used to periodically check for incoming packets."

Reply:
I did threadsafe test and confirmed that

Reply:
OK, so basically their implementation does not support non-blocking sockets.